### PR TITLE
common: Implemented thread safe version of dlist

### PR DESCRIPTION
**Pull request by dmitrygladkov**
`struct dlist_ts` - container for lock and dlist
All functions take `head` as an argument to be able to insert, start a loop and so on not only to/from real headd of dlist, but to/from any element.

Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

**Comment by dmitrygladkov** [comment]
@shefty this addresses your comment from #4212

File: include/ofi_list.h [hunk 1]
```diff
@@ -194,6 +194,124 @@ static inline void dlist_splice_tail(struct dlist_entry *head,
...
+ */
+struct dlist_mt {
+	struct dlist_entry	*head;
+	fastlock_t		lock;
+};
+
+static inline void dlist_mt_init(struct dlist_mt *list)
+{
+	fastlock_init(&list->lock);
+	list->head = NULL;
```
Comment (shefty) [inline include/ofi_list.h]: dlist_init

File: include/ofi_list.h [hunk 2]
```diff
@@ -194,6 +194,124 @@ static inline void dlist_splice_tail(struct dlist_entry *head,
...
+}
+
+static inline int dlist_mt_empty(struct dlist_mt *list)
+{
+	return list->head == NULL;
```
Comment (shefty) [inline include/ofi_list.h]: dlist_empty

File: include/ofi_list.h [hunk 3]
```diff
@@ -194,6 +194,124 @@ static inline void dlist_splice_tail(struct dlist_entry *head,
 	dlist_splice_head(head->prev, to_splice);
 }
 
+/*
+ * Multi-threaded Double-linked list
+ */
+struct dlist_mt {
```
Comment (shefty) [inline include/ofi_list.h]: Do we like mt for multi-threaded or ts for thread safe better?

File: include/ofi_list.h (hunk 2 above)
Comment (dmitrygladkov) [reply include/ofi_list.h]: Done

File: include/ofi_list.h (hunk 1 above)
Comment (dmitrygladkov) [reply include/ofi_list.h]: Done

File: include/ofi_list.h (hunk 3 above)
Comment (a-ilango) [reply include/ofi_list.h]: +1 for thread safe

**Pull request merged**